metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -1884,13 +1892,17 @@ private UpdateAspectResult ingestAspectToLocalDBNoTransaction(
       @Nonnull final AuditStamp auditStamp,
       @Nonnull final SystemMetadata providedSystemMetadata,
       @Nullable final EntityAspect latest,
-      @Nonnull final Long nextVersion) {
+      @Nonnull final Long nextVersion,
+      @Nullable final ExtraIngestParams extraIngestParams) {
 
-    // 2. Compare the latest existing and new.
+    // 1. Compare the latest existing and new.
     final RecordTemplate oldValue =
         latest == null ? null : EntityUtils.toAspectRecord(urn, aspectName, latest.getMetadata(), getEntityRegistry());
     final RecordTemplate newValue = updateLambda.apply(Optional.ofNullable(oldValue));
 
+    // 2. Check if the condition update is met
+    checkConditionalUpdate(latest, extraIngestParams);
```

Comment:
You could probably push this check further up (above the `final RecordTemplate newValue` line) as we can fail faster that way.